### PR TITLE
Storybook config updates

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

- TSDocs for Button, Callout, etc
- Change DarkModeWrapper to cover entire iframe
- Add `defaultArgTypes` to define reusable argTypes for common props
- Bump Storybook to 6.5
- Use `BaseFontSize` from tokens in Button and Callout
- Change Storybook to Docs mode. Wanted to try this out to see how the team likes it.

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -113,7 +130,9 @@ module.exports = {
     },
     {
       files: ['website/**/*.{ts,tsx}'],
-      rules: { 'react/react-in-jsx-scope': 'off' },
+      rules: {
+        'react/react-in-jsx-scope': 'off',
+      },
     },
     {
       files: ['packages/**/*.spec.{ts,tsx}'],
```
Comment (spark33) [inline .eslintrc.js:138]: The props are not changing - the Storybook eslint plugin is just changing the formatting for some reason.

File: package.json [hunk 2]
```diff
@@ -142,6 +142,7 @@
     "eslint-plugin-jsx-a11y": "^6.3.1",
     "eslint-plugin-react": "7.21.5",
     "eslint-plugin-react-hooks": "^4.2.0",
+    "eslint-plugin-storybook": "^0.5.12",
```
Comment (spark33) [inline package.json:145]: Added automatically by `npx` script when bumping Storybook to 6.5.

File: packages/marketing-modal/src/MarketingModal.story.tsx [hunk 3]
```diff
@@ -47,9 +56,23 @@ export default {
...
+        <>
+          <Title />
+          <SbSubtitle />
+          <Description />
+          <Primary />
+          <ArgsTable story={PRIMARY_STORY} />
+        </>
+      ),
+    },
+  },
```
Comment (spark33) [inline packages/marketing-modal/src/MarketingModal.story.tsx:72]: Remove since Uncontrolled was removed

**Review by shaneeza — APPROVED** [review]

File: .storybook/preview.js [hunk 4]
```diff
@@ -10,12 +10,20 @@ import {
...
+export const argTypes = {
+  className: {
+    description: '`className` prop passed to the component',
+    control: 'text',
+  },
+};
+
 export const parameters = {
   actions: { argTypesRegex: '^on[A-Z].*' },
+  viewMode: 'docs',
```
Comment (spark33) [inline .storybook/preview.js:22]: Remove - might not be helpful for devs

**Pull request merged**